opentelemetry-otlp/tests/integration_test/tests/metrics.rs

```diff
@@ -0,0 +1,23 @@
+use std::{fs::File, io::Write};
+
+use integration_test_runner::metrics_asserter::{read_metrics_from_json, MetricsAsserter};
+use opentelemetry_proto::tonic::metrics::v1::MetricsData;
+
+#[test]
+fn test_serde() {
+    let metrics = read_metrics_from_json(File::open("./expected/metrics.json").unwrap());
+
+    let json = serde_json::to_string_pretty(&MetricsData {
+        resource_metrics: metrics,
+    })
+    .expect("Failed to serialize metrics");
+
+    // Write to file.
+    let mut file = File::create("./expected/serialized_metrics.json").unwrap();
```

Comment:
We can use [tempfile](https://crates.io/crates/tempfile) here

Reply:
I am not sure we should, leaving the expected shape here committed to source allows us to almost use a snapshot-testing technique. 

The problem is, if the serde logic is wrong, then the serialization from `./expected/metrics.json` would be wrong, and the deserialization back into a file would also be wrong, **but because both the serde is wrong in the same manner, the `MetricsAsserter` would assert their shapes to be the same**. By leaving the expected file committed to source, we'll be able to notice that the actual errors in serde.
- For example, try commenting out all the changes in `grpc_build.rs` in this PR, generating the proto files once, and then running the metrics integration test introduced in this PR. I also logged the `left` and `right` and I can see that they are both incorrectly serde **in the same manner** (`data: None`):

```
left: [
    ResourceMetrics {
        resource: Some(
            Resource {
                attributes: [
                    KeyValue {
                        key: "service.name",
                        value: Some(
                            AnyValue {
                                value: Some(
                                    StringValue(
                                        "my.service",
                                    ),
                                ),
                            },
                        ),
                    },
                ],
                dropped_attributes_count: 0,
            },
        ),
        scope_metrics: [
            ScopeMetrics {
                scope: Some(
                    InstrumentationScope {
                        name: "my.library",
                        version: "1.0.0",
                        attributes: [
                            KeyValue {
                                key: "my.scope.attribute",
                                value: Some(
                                    AnyValue {
                                        value: Some(
                                            StringValue(
                                                "some scope attribute",
                                            ),
                                        ),
                                    },
                                ),
                            },
                        ],
                        dropped_attributes_count: 0,
                    },
                ),
                metrics: [
                    Metric {
                        name: "my.counter",
                        description: "I am a Counter",
                        unit: "1",
                        metadata: [],
                        data: None,
                    },
                    Metric {
                        name: "my.gauge",
                        description: "I am a Gauge",
                        unit: "1",
                        metadata: [],
                        data: None,
                    },
                    Metric {
                        name: "my.histogram",
                        description: "I am a Histogram",
                        unit: "1",
                        metadata: [],
                        data: None,
                    },
                ],
                schema_url: "whatever",
            },
        ],
        schema_url: "whatever",
    },
]
right: [
    ResourceMetrics {
        resource: Some(
            Resource {
                attributes: [
                    KeyValue {
                        key: "service.name",
                        value: Some(
                            AnyValue {
                                value: Some(
                                    StringValue(
                                        "my.service",
                                    ),
                                ),
                            },
                        ),
                    },
                ],
                dropped_attributes_count: 0,
            },
        ),
        scope_metrics: [
            ScopeMetrics {
                scope: Some(
                    InstrumentationScope {
                        name: "my.library",
                        version: "1.0.0",
                        attributes: [
                            KeyValue {
                                key: "my.scope.attribute",
                                value: Some(
                                    AnyValue {
                                        value: Some(
                                            StringValue(
                                                "some scope attribute",
                                            ),
                                        ),
                                    },
                                ),
                            },
                        ],
                        dropped_attributes_count: 0,
                    },
                ),
                metrics: [
                    Metric {
                        name: "my.counter",
                        description: "I am a Counter",
                        unit: "1",
                        metadata: [],
                        data: None,
                    },
                    Metric {
                        name: "my.gauge",
                        description: "I am a Gauge",
                        unit: "1",
                        metadata: [],
                        data: None,
                    },
                    Metric {
                        name: "my.histogram",
                        description: "I am a Histogram",
                        unit: "1",
                        metadata: [],
                        data: None,
                    },
                ],
                schema_url: "whatever",
            },
        ],
        schema_url: "whatever",
    },
]
```

Reply:
@hwrdtm I understand that if the serde logic is incorrect, serialization and deserialization would be incorrect in same way, and the actual error would get unnoticed. However, how is the committing the expected output in git helping here, as we are not using it for comparison anywhere? Probably I am missing something :)

Reply:
It produces a git diff that the author would hopefully notice and dig into. It's not foolproof, but is consistent with snapshot testing paradigms, because sometimes we may actually wish to change the snapshot, and so the git diff is not always necessarily an error, and human judgement is helpful here.

Reply:
Thanks for clarification.